File: src/EzBud.Domain/Account/Account.cs
namespace EzBud.Domain.Account;

public class Account(Guid id, string name, bool onBudget = true)
{
    public const int MaxNameLength = 70;

    public Guid Id { get; set; } = id;
    public string Name { get; set; } = name;
    public bool OnBudget { get; set; } = onBudget;
    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}

File: src/main/java/dk/michaelbui/ezbud_server/domain/model/Account.java
package dk.michaelbui.ezbud_server.domain.model;

import java.util.ArrayList;
import java.util.Collection;
import java.util.UUID;

public class Account {
    private UUID id = UUID.randomUUID();
    private String name;
    private boolean isOnBudget;
    private Collection<Transaction> transactions = new ArrayList<>();

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isOnBudget() {
        return isOnBudget;
    }

    public void setOnBudget(boolean onBudget) {
        isOnBudget = onBudget;
    }

    public Collection<Transaction> getTransactions() {
        return transactions;
    }

    public void setTransactions(Collection<Transaction> transactions) {
        this.transactions = transactions;
    }
}
